model/database/spans.yaml

```diff
@@ -573,6 +570,12 @@ groups:
         examples: ["200", "201"]
         requirement_level:
           conditionally_required: if response was received
+      - ref: db.response.returned_rows
+        brief: >
+          Cosmos DB row count in result set.
+        examples: [10, 20]
+        requirement_level:
+          conditionally_required: if response was received and returned any rows
```

Comment:
@lmolkova should we also add to database spans in this PR (as experimental, not RC)?

the reason I'm wondering is because I'm not sure this "conditionally_required" definition will work in the general case

and then we might want to consider how to differentiate between:
* zero rows were returned
* the instrumentation doesn't support capturing this attribute

I'm also ok making that a future problem (outside the scope of this PR), though I do think we would want to figure it out before we mark this attribute as stable (i.e. used in any stable cosmosdb semconv)

Reply:
@trask  may be you can create a github item to discuss further on this separately. :)

Reply:
unresolving this discussion to give @lmolkova a chance to review

Reply:
It'd be great to add this attribute to the general `trace.db.common.query_and_collection` group and say it's recommended there. It's effectively 1 line change and since it's a generic attribute, it makes sense to reference it on the general group.

Reply:
> we might want to consider how to differentiate between:
> 
> * zero rows were returned
> * the instrumentation doesn't support capturing this attribute

opened an issue for this: https://github.com/open-telemetry/semantic-conventions/issues/1522